common/os_calls.c

```diff
@@ -2822,6 +2822,22 @@ g_pos(const char *str, const char *to_find)
     return (pp - str);
 }
 
+/*****************************************************************************/
+int
+g_last_ch_pos(const char *str, char ch_to_find, int n)
```

Comment:
you want `strrchr` from `string.h`

then do pointer arithmetic:
```c
e = strchr(string, 'e');
index = (int)(e - string);
```

Reply:
No I can't do this. I would not like to trust on the input beeing correctly terminated with \0. Just like the the code I'd like to patch.
Unfortunately the C standard does not know of strnrchr() so I had to write my own...